### PR TITLE
CUMULUS-811: Preserve S3 Object Tags

### PR DESCRIPTION
**Pull request by mennovandiermen**
**Summary:** Summary of changes

Addresses [CUMULUS-811: Preserve S3 object tags](https://bugs.earthdata.nasa.gov/browse/CUMULUS-XXX)

## Changes

* Detailed list or prose of changes
* ...

## Test Plan
Things that should succeed before merging.

- [x] Unit tests
- [x] Adhoc testing
- [x] Update CHANGELOG

## Release PR

If this is a **release** PR, make sure you branch name starts with `release-` prefix, e.g. `release-v-1.1.2`.

Reviewers: @tag-your-friends

### CODE REVIEW

File: example/spec/ingestGranule/IngestGranuleSuccessSpec.js [hunk 1]
```diff
@@ -93,6 +98,7 @@ describe('The S3 Ingest Granules workflow', () => {
   let inputPayload;
   let expectedSyncGranulePayload;
   let expectedPayload;
+  let expectedS3TagSet;
```
[the remark this answers is not in the document]
Comment (mennovandiermen) [reply example/spec/ingestGranule/IngestGranuleSuccessSpec.js]: It's actually used in the test set up to tag the provider files, so we can end-to-end test tag preservation.

File: tasks/move-granules/index.js [hunk 2]
```diff
@@ -269,19 +269,24 @@ async function updateCmrFileAccessURLs(cmrFiles, granulesObject, allFiles, distE
     cmrFile.metadata = xml;
     /* eslint-enable no-param-reassign */
     const updatedCmrFile = granule.files.find((f) => f.filename.match(/.*\.cmr\.xml$/));
+    // S3 upload only accepts tag query strings, so reduce tags to query string.
+    const tagsQueryString = cmrFile.s3Tags.reduce((acc, tag) => acc.concat(`&${tag.Key}=${tag.Value}`), '').substring(1);
```
[the remark this answers is not in the document]
Comment (mennovandiermen) [reply tasks/move-granules/index.js]: Good point

File: packages/ingest/aws.js [hunk 3]
```diff
@@ -152,6 +152,8 @@ class Events {
 
 class S3 {
   static parseS3Uri(uri) {
+    // eslint-disable-next-line max-len
+    log.warn('@cumulus/ingest/aws/S3.parseUri is deprecated.  Use @cumulus/common/aws.parseS3Uri instead.');
```
[the remark this answers is not in the document]
Comment (mennovandiermen) [reply packages/ingest/aws.js]: Can do

File: packages/ingest/aws.js [hunk 4]
```diff
@@ -486,7 +502,7 @@ class KMS {
 class StepFunction {
   static async getExecution(arn, ignoreMissingExecutions = false) {
     // eslint-disable-next-line max-len
-    log.debug('@cumulus/ingest/aws/StepFunction.getExecution is deprecated.  Use @cumulus/common/step-functions/describeExecution instead.');
+    log.debug('@cumulus/ingest/aws/StepFunction.getExecution is deprecated as of Cumulus v1.10.2.  Use @cumulus/common/step-functions/describeExecution instead.');
```
[the remark this answers is not in the document]
Comment (mennovandiermen) [reply packages/ingest/aws.js]: I hadn't intended to touch that line, but I guess I replace-all'd it anyway. I'll change it to a warn, too.

File: packages/ingest/crypto.js [hunk 5]
```diff
@@ -26,7 +27,9 @@ class S3KeyPairProvider {
     const pki = forge.pki;
     const b = bucket || process.env.internal;
     const s = stack || process.env.stackName;
-    const pub = await S3.get(b, `${s}/crypto/${keyId}`);
+    const pub = await s3().getObject({
```
Comment (laurenfrederick) [inline packages/ingest/crypto.js]: Nitpicky, but the variables could be named better.

**Pull request merged**